tests/Aspire.MongoDB.Driver.Tests/MongoDbContainerFixture.cs

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Components.Common.Tests;
+using MongoDB.Bson;
+using MongoDB.Driver;
+using Testcontainers.MongoDb;
+using Xunit;
+
+namespace Aspire.MongoDB.Driver.Tests;
+
+public sealed class MongoDbContainerFixture : IAsyncLifetime
+{
+    public MongoDbContainer? Container { get; private set; }
+
+    public string GetConnectionString() => Container?.GetConnectionString() ??
+        throw new InvalidOperationException("The test container was not initialized.");
+
+    public async Task InitializeAsync()
+    {
+        if (RequiresDockerTheoryAttribute.IsSupported)
+        {
+            Container = new MongoDbBuilder()
+                .WithImage("mongo:7.0.5")
+                .Build();
+            await Container.StartAsync();
+
+            // Create `test_db` database with user:mongo pwd:mongo
+            var mongoClient = new MongoClient(Container.GetConnectionString());
+            var createUserCommand = new BsonDocumentCommand<BsonDocument>(BsonDocument.Parse("""
+            {
+               createUser: "mongo",
+               pwd: "mongo",
+               roles: [ { role: 'readWrite', db: 'test_db' } ]
+            }
+            """));
+            await mongoClient.GetDatabase("test_db")
+                .RunCommandAsync(createUserCommand);
```

Comment:
In our "playground" app, we just do:

https://github.com/dotnet/aspire/blob/ded1eaef1592000b7c99136d1cc9a9085170f07b/playground/mongo/Mongo.ApiService/Program.cs#L19-L20

Is that enough? That seems simpler than what we have here.

Reply:
With the only the two lines above, it still fails with:
`,"msg":"Failed to authenticate","attr":{"client":"172.17.0.1:58864","isSpeculative":false,"isClusterMember":false,"mechanism":"SCRAM-SHA-1","user":"mongo","db":"test_db","error":"UserNotFound: Could not find user \"mongo\" for db \"test_db\"",`

The testcontainers use `mongo:mongo` by default.
How is the container setup/started in case of the playground?

Reply:
> How is the container setup/started in case of the playground?

https://github.com/dotnet/aspire/blob/ded1eaef1592000b7c99136d1cc9a9085170f07b/playground/mongo/Mongo.AppHost/Program.cs#L6

Reply:
If the default is to start it without any user/pass then we can do the same in tests too.
```csharp
            Container = new MongoDbBuilder()
                .WithImage("mongo:7.0.5")
                .WithUsername(null)
                .WithPassword(null)
                .Build();
```